libp2p/pubsub/pubsub.py

```diff
@@ -417,10 +438,14 @@ def handle_subscription(
         logger.debug("unsubscribing from topic %s", topic_id)
 
         # Return if we already unsubscribed from the topic
-        if topic_id not in self.my_topics:
+        if topic_id not in self.topic_ids:
             return
-        # Remove topic_id from map if present
-        del self.my_topics[topic_id]
+        # Remove topic_id from the maps before yielding
+        send_channel = self.subscribed_topics_send[topic_id]
+        del self.subscribed_topics_send[topic_id]
+        del self.subscribed_topics_receive[topic_id]
+        # Only close the send side
+        await send_channel.aclose()
```

Comment:
My intuition (which very well might be wrong) is that this should happen before removing the channels from the maps..